osquery/core/database/rocksdb_migration.cpp

```diff
@@ -0,0 +1,228 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/core/conversions.h>
+#include <osquery/core/database/rocksdb_migration.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+namespace osquery {
+
+namespace fs = boost::filesystem;
+
+ExpectedSuccess<RocksdbMigrationError> RocksdbMigration::migrateDatabase(
+    const std::string& path) {
+  auto migration = std::make_unique<RocksdbMigration>(path);
+  return migration->migrateIfNeeded();
+}
+
+RocksdbMigration::RocksdbMigration(const std::string& path) {
+  auto boost_path = fs::path(path).make_preferred();
+  source_path_ = boost_path.string();
+}
+
+ExpectedSuccess<RocksdbMigrationError> RocksdbMigration::migrateIfNeeded() {
+  auto open_result = openDatabase(source_path_, false, false);
+  if (open_result) {
+    auto db_handle = open_result.take();
+    auto version_result = getVersion(db_handle);
+    if (version_result) {
+      int version = version_result.take();
+      if (version != kDatabaseSchemaVersionCurrent) {
+        migrateFromVersion(version);
+      } else {
+        return Success();
+      }
+    } else {
+      return version_result.takeError();
+    }
+  } else {
+    // InvalidArgument means that db does not exists
+    if (open_result.getErrorCode() == RocksdbMigrationError::InvalidArgument) {
+      return Success();
+    } else {
+      return open_result.takeError();
+    }
+  }
+  return Success();
+}
+
+ExpectedSuccess<RocksdbMigrationError> RocksdbMigration::dropDbMigration(
```

Comment:
Whats that function for?
As I understand it creates or fails. But no use, and not sure if function name expresses what it does

Reply:
It will be used in future, as builtin migration for cases when we would not support migration from particular versions